src/plugins/hidsim/hid_injection.cpp

```diff
@@ -204,14 +209,30 @@ static int get_display_dimensions(qmp_connection* qc, dimensions* dims)
     int matches = 0;
     int ret = 0;
 
-    /* Checks PPM-header */
-    matches = fscanf(fr, "%s", ppm_hdr);
+    /* Reads first three bytes of the potential portable pixmap */
+    matches = fscanf(fr, "%2s", ppm_hdr);
 
+    /* Checks PPM-header */
     if (matches != 1 || strcmp(ppm_hdr, PPM_MAGIC_NUMBER) != 0)
+    {
+        fclose(fr);
         ret = -1;
+    }
 
     /* Reads actual dimensions */
-    matches = fscanf(fr, "%d%d", &(dims->width), &(dims->height));
+    uint32_t width = 0, height = 0;
+    matches = fscanf(fr, "%" SCNd32 "%" SCNd32, &width, &height);
+
+    /* Sanitizes read values */
+    dims->width = width > UINT32_MAX ? UINT32_MAX : width;
+    dims->height = height > UINT32_MAX ? UINT32_MAX : height;
```

Comment:
Code scanning

Comparison result is always the same
Comparison is always false because height <= 4294967295

Reply:
Changed to `UINT16_MAX` and forcely pushed the change. Sorry for the negligence!!

Reply:
Thanks for reviewing!